### PR TITLE
Add devcontainer for Codespaces

### PR DESCRIPTION
**Pull request by perlpunk**
Issue: https://progress.opensuse.org/issues/87695

### CODE REVIEW

**Comment by perlpunk** [comment]
Needs 
* https://github.com/os-autoinst/os-autoinst-common/pull/52

**Comment by perlpunk** [comment]
Needs
* https://github.com/os-autoinst/os-autoinst-distri-example/pull/42

**Review by kalikiana — COMMENTED** [review]
Very cool. Gave it a go and it immediately offered to open the web UI. Not sure what's with the postCommand taking so long 🤔 Seems stuck for minutes without any feedback besides an animated icon.

image

The terminal is effectively blocked. I can't check the logfile.

**Comment by perlpunk** [comment]
You need to open another terminal yourself to be able to check the logfile

**Comment by perlpunk** [comment]
The postCreateCommand is openqa-bootstrap, and that never stops.

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    "name": "openQA Single Instance",
+    "image": "registry.opensuse.org/devel/openqa/containers15.5/openqa-single-instance",
+    "runArgs": [ "--privileged", "--device", "/dev/kvm", "--entrypoint", "bash" ],
+    "postCreateCommand": "chown root:kvm /dev/kvm",
+    "containerEnv": {
+      "VNCPORT_OFFSET": "100"
+    },
+    "postStartCommand": "/usr/share/openqa/script/openqa-bootstrap >/var/log/openqa-bootstrap.log 2>&1"
```
Comment (kalikiana) [inline .devcontainer/devcontainer.json:9]: I found an example which spawns a node application https://github.com/blackgirlbytes/codespaces-forward-port

The web app responds to ^C and I see a "bash" tab.

image

This is clear enough for me. I'm wondering why we don't see this in the openQA setup

Comment (kalikiana) [reply .devcontainer/devcontainer.json:9]: Also in this node example app it tells me what it's doing and where the server is running (it's not fully visible in the screenshot).

Comment (kalikiana) [reply .devcontainer/devcontainer.json:9]: I guess [it's using](https://github.com/blackgirlbytes/codespaces-forward-port/blob/main/.devcontainer/devcontainer.json#L9) `postAttachCommand` and maybe that makes a difference?

Comment (perlpunk) [reply .devcontainer/devcontainer.json:9]: maybe I need to use `postAttachCommand` instead

Comment (kalikiana) [reply .devcontainer/devcontainer.json]: Nice! Much better now. Thank you for making those tweaks 🥰

**Pull request merged**